### DISCUSSION TITLE
Can't printscreen on wayland

### DISCUSSION THREAD

**Discussion by arw357** [regolith-linux/regolith-desktop#1009] [Q&A]
The printscreen button doesn't work  I'm ok with this. 
But also flameshot doesn't work and it used to work recently.  
```
station1 xdg-desktop-por[4759]: Unhandled parent window type 
mai 13 16:07:25 station1 xdg-desktop-por[4759]: Failed to associate portal window with parent window ''
mai 13 16:07:25 station1 xdg-desktop-portal-gnome[4759]: Failed to get screenshot: Cannot invoke method; proxy is for the well-known name org.gnome.Shell.Screenshot without an owner, and proxy was constructed with the G_DBUS_PROXY_FLAGS_DO_NOT_AUTO_START flag

```
Where should I look ?

**Comment by ajkerrigan** [comment]
Hey there, I ran into the same issue. Flameshot's [Wayland Help](https://flameshot.org/docs/guide/wayland-help/) sounded promising and linked to relevant tracking issues, but none of the workarounds seemed to help in my case.

In the meantime I created `~/.config/regolith3/sway/config.d/90_screenshot` containing:

```
bindsym Print exec "slurp | grim -g - - | wl-copy"
```

And now printscreen lets me draw a rectangle and snag a screen section to the clipboard. I'd still prefer Flameshot of course, but in a pinch editing things in gimp still works :shrug: . Hope some of this helps.

(Note: You may need to `sudo apt install slurp grim wl-clipboard` if you're missing any of those packages.)

**Comment by SoumyaRanjanPatnaik** [comment]
I haven't been able to settle on good defaults for screenshots. We probably need to think about the following before deciding on a solution.
- Select a good screenshot utility. `grimshot` is currently my top pick. 
- Default mode (on PrintScr): Fullscreen, window selection or area
- Are keybindings required for non-default modes as well?
- Save to file or copy to clipboard

Most of these issues would be solved by a GUI screenshot utility, but I'm not aware of one which is lightweight and stable on wayland.

**Reply by cfsmp3** [reply]
I'm partial to flameshot, *but* it has a bug in sway in which only save to disk works (not clipboard), and they are so slow with bugfixes and PRs that I'm not sure I'll ever be fixed. On the other hand, I like that it's cross platform and quite powerful (for a screen capture tool).

I prefer area selection over full screen. I hardly ever want to copy or share my full screen.

I wouldn't add more keybindings. If someone has a need, they can just add their own :-)

flameshot lets you decide what to do as part of its interface. If we had no interface (don't now how grimshot work) and we had to select one default action... I guess it depends on the area decision. If it's full screen, then it should go to the clipboard (because some edit will be needed by a different tool), if it's a screen region, I guess... it's 50%-50%

**Reply by kgilmer** [reply]
* I tried flameshot, gnome-screenshot, and grimshot.  I like grimshot.  Flameshot is feature rich and I can see the appeal, but I experienced some weirdness with it.  I prefer simple tools.
* I came up with the following scheme
** `<super><print>` puts the active window's bitmap into the clipboard.  This is the most "minimal" configuration and I think this will be by far the most common use case, and by design does not write to the filesystem. 
** Adding the `<Shift>` modifier enables grimshot's "anything" mode which allows the user to select what they want to copy and then writes to file, to clipboard, and generates a notification.
** There is a special case that "anything" mode doesn't capture: "everything", and so adding to the keybinding `<Ctrl>` writes all outputs to disk and generates a notification.  I see this useful in cases there is something on a screen the user wants to capture ASAP.  

PR for review, tested against 3.2 beta 2: https://github.com/regolith-linux/regolith-wm-config/pull/38

**Reply by kgilmer** [reply]
IMO full screen (or even worse, all screens) screen captures that automatically write the image to disk is a security risk, and so desire to make them harder to accidentally trigger.  IMO writing a specific window to the clipboard buffer is the safest mode.

**Reply by SoumyaRanjanPatnaik** [reply]
This sounds good for now. Although I do eventually want to work on a simple wrapper around grimshot that allows the user to select which mode they want to capture and what they want to do with the image.

**Reply by kgilmer** [reply]
> ..allows the user to select which mode they want to capture and what they want to do with the image.

Are you thinking like some OSD overlay menu like flameshot? Something else?

**Reply by kgilmer** [reply]
PR mentioned above is now built into `unstable` (FYI)
